libraries/botframework-schema/src/index.ts

```diff
@@ -17,939 +19,1122 @@ export * from './teams';
  * Attachment View name and size
  */
 export interface AttachmentView {
-  /**
-   * Id of the attachment
-   */
-  viewId: string;
-  /**
-   * Size of the attachment
-   */
-  size: number;
+    /**
+     * Id of the attachment
+     */
+    viewId: string;
+    /**
+     * Size of the attachment
+     */
+    size: number;
 }
 
+export const assertAttachmentView: Assertion<AttachmentView> = (value, path) => {
+    assert.unsafe.objectAs<AttachmentView>(value, path);
+    assert.string(value.viewId, path.concat('viewId'));
+    assert.number(value.size, path.concat('size'));
+};
+
+const assertAttachmentViewArray: Assertion<Array<AttachmentView>> = assert.arrayOf(assertAttachmentView);
+
+export const isAttachmentView = assert.toTest(assertAttachmentView);
+
 /**
  * Metadata for an attachment
  */
 export interface AttachmentInfo {
-  /**
-   * Name of the attachment
-   */
-  name: string;
-  /**
-   * ContentType of the attachment
-   */
-  type: string;
-  /**
-   * attachment views
-   */
-  views: AttachmentView[];
-}
+    /**
+     * Name of the attachment
+     */
+    name: string;
+    /**
+     * ContentType of the attachment
+     */
+    type: string;
+    /**
+     * attachment views
+     */
+    views: AttachmentView[];
+}
+
+export const assertAttachmentInfo: Assertion<AttachmentInfo> = (value, path) => {
+    assert.unsafe.objectAs<AttachmentInfo>(value, path);
```

Comment:
This is the usage I've alluded to [here](https://github.com/microsoft/botbuilder-js/pull/3066/files#diff-9b546a3a8b1030340c9c20edc96fc3f7bb3b3dd69c942d7b8b393f16a5727d3bR138).